number_parser/parser.py

```diff
@@ -148,6 +153,7 @@ def parse(input_string):
 
```

Comment:
Pardon me but I didn't understand the meaning of `tokens_takens` Does it mean tokens used so far?